pod/perldelta.pod

```diff
@@ -36,6 +36,62 @@ distinctions, this can be useful when serialising or encoding Perl data
 structures for export to other languages or data formats where such a
 distinction does matter.
 
+=head2 Support for not sharing all keys in large hash tables.
+
+This release includes support to build Perl so that it does not share
+all keys in hash tables as has been historically the case since Perl
+v5.6. The shared key behavior was added to Perl originally to support
+having many objects with the same keys without having to store each key
+many times. For objects that might be created hundreds or thousands of
+times this key sharing behavior can result in using significantly less
+memory. However key sharing comes at the cost of having to maintain an
+internal "master hash table" (PL_str_tab), and having to do two store
+operations per key stored in a hash table. It also can result in
+surprising consequences with programs that fork, causing the master hash
+table to be COWed (copy on write) into each forked processes memory
+space which can, in worse case, greatly increase the memory utilization
+of such scripts.
+
+The new behavior is an attempt to have our cake and eat it too. Small
+hash tables will continue to share keys, but once the hash table is
+used to store more than a certain number of keys (currently 64) it will
+not share the new keys that are added. This should benefit scripts that
+build large hash tables for aggregation purposes, especially those that
+do so in forked sub-processes. The unshared mode is triggered when the
+hash table is resized which is non-deterministic and depends on the
+underlying hash function that perl has been built with and the seed
+used by the current process so the exact circumstances when this new
+mode will be enabled for a given hash will vary between different
+invocations of a script.
+
+We are uncertain of the exact effects of this new mode and are keen to
+get field reports about the consequences of enabling it. We know that in
+many cases it speeds up code that builds large hash tables. We also know
+that in some extreme situations it can significantly increase memory
+consumption. We also have data to suggest that it will reduce COW churn
+in scripts that use fork and build large hash tables in child processes
+for aggregation purposes. There is also data which suggests that in some
```

Comment:
Same phrase "for aggregation purposes" here: I still don't really know what you mean by it, but for this context it is fine to talk about the purpose if it is reasonably obvious what kind of mechanics it implies.